### PR TITLE
CI: add support for new cri-o job

### PR DESCRIPTION
**Pull request by chavafg**
This change is needed because now that we plan to test
the changes in the cri-o repository, the CI needs to know
if the change being tested comes from the cri-o repository
or from our kata-containers repos. If it comes from the cri-o
repository, we should not use the version we support on our
versions.yaml. We should stick to the changes in the PR.

Fixes: #230.

Signed-off-by: Salvador Fuentes <salvador.fuentes@intel.com>

### CODE REVIEW

File: .ci/install_crio.sh [hunk 1]
```diff
@@ -12,11 +12,14 @@ source "${cidir}/lib.sh"
...
-crio_version=$(get_version "externals.crio.version")
 go get -d "$crio_repo" || true
 pushd "${GOPATH}/src/${crio_repo}"
-git fetch
-git checkout "${crio_version}"
+
+if [ "$ghprbGhRepository" != "${crio_repo/github.com\/}" ];
+	crio_version=$(get_version "externals.crio.version")
+	git fetch
+	git checkout "${crio_version}"
```
Comment (sboeuf) [inline .ci/install_crio.sh]: We need to test a specific PR, we're actually not testing master, and we're not testing a specific version either.

Comment (chavafg) [reply .ci/install_crio.sh]: @sboeuf that is right, but I am checking-out to the specific PR on the jenkins job configuration. Which does this:
```
crio_repo="github.com/kubernetes-incubator/cri-o"
tests_repo="github.com/kata-containers/tests"
go get -d "$crio_repo" || true
go get -d "$tests_repo" || true

cd "${GOPATH}/src/${crio_repo}"

git fetch origin "pull/${pr_number}/head:${pr_branch}"
git checkout "${pr_branch}"
git rebase "origin/${ghprbTargetBranch}"
```

And this line only checkout to the `$crio_version` on our `versions.yaml` only if the the PR does not come from the `kubernetes-incubator/cri-o` repository.

Comment (sboeuf) [reply .ci/install_crio.sh]: Yep definitely, I have read the code in the wrong direction, sorry...

File: .ci/install_crio.sh [hunk 2]
```diff
@@ -12,11 +12,14 @@ source "${cidir}/lib.sh"
 
 echo "Get CRI-O sources"
 crio_repo="github.com/kubernetes-incubator/cri-o"
-crio_version=$(get_version "externals.crio.version")
 go get -d "$crio_repo" || true
 pushd "${GOPATH}/src/${crio_repo}"
-git fetch
-git checkout "${crio_version}"
+
+if [ "$ghprbGhRepository" != "${crio_repo/github.com\/}" ];
```
Comment (sboeuf) [inline .ci/install_crio.sh]: you forgot a `then`

Comment (chavafg) [reply .ci/install_crio.sh]: fixed, thanks.

**Comment by chavafg** [comment]
@grahamwhaley took a look at the CI jobs and it gets the correct cri-o version. From 
http://kata-jenkins-ci.westus2.cloudapp.azure.com/job/kata-containers-tests-ubuntu-17-10-PR/86/consoleFull 

```
Note: checking out 'v1.9.10'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 87237324... version: v1.9.10
```

**Comment by GabyCT** [comment]
lgtm

**Comment by sboeuf** [comment]
@chavafg Now that we got this merged, let's sync on Monday and talk with CRI-O maintainers about enabling the Github webhook on their side.
And we'll need to submit some test PRs on their repo in order to verify that an empty PR works as expected and that we get CRIO testing passing with Kata.
